### PR TITLE
#2616 Adding operating point B, which is stable

### PR DESCRIPTION
**Pull request by ErwanGuichard49**
closes [2616](https://github.com/dynawo/dynawo/issues/2616)

### CODE REVIEW

**Review by adriguir — COMMENTED** [review]
At first glance, the development seems globally fine and is done in a very nice way. There is only one modification missing: the active power of generator g20 should be decreased by 600 MW to balance the addition of the new g16b generator, which is not the case.
It ends up in a non flat trajectory even without events.

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Grid/FullDynamicModelB.mo [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+*/
+
+model FullDynamicModelB "Nordic test grid with buses, lines, shunts, voltage-dependent loads, transformers and generators (operating point B)"
+  import Modelica.SIunits;
+  import Dynawo;
+  import Dynawo.Examples.Nordic.Components.GeneratorWithControl;
+  import Dynawo.Types;
+
+  extends Dynawo.Examples.Nordic.Grid.FullDynamicModelA(
+    g15(P0Pu = P0Pu_g15a, Q0Pu = Q0Pu_g15a),
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Grid/FullDynamicModelB.mo:23]: Check gen =

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Grid/FullDynamicModelB.mo [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  Dynawo.Electrical.Transformers.TransformerFixedRatio trafo_g15_4047b(BPu = 0, GPu = 0, RPu = 0, XPu = 0.15 * 1.05 ^ 2 * (100 / 600.0), rTfoPu = 1.05) annotation(
+    Placement(visible = true, transformation(origin = {95, -103}, extent = {{-5, -5}, {5, 5}}, rotation = -90)));
+  Dynawo.Electrical.Transformers.TransformerFixedRatio trafo_g16_4051b(BPu = 0, GPu = 0, RPu = 0, XPu = 0.15 * 1.05 ^ 2 * (100 / 700.0), rTfoPu = 1.05) annotation(
+    Placement(visible = true, transformation(origin = {5, -137}, extent = {{5, -5}, {-5, 5}}, rotation = -90)));
+  Dynawo.Electrical.Transformers.TransformerFixedRatio trafo_g18_4063b(BPu = 0, GPu = 0, RPu = 0, XPu = 0.15 * 1.05 ^ 2 * (100 / 600.0), rTfoPu = 1.05) annotation(
+    Placement(visible = true, transformation(origin = {-110, -137}, extent = {{-5, -5}, {5, 5}}, rotation = 90)));
+
+  GeneratorWithControl.GeneratorSynchronousFourWindingsWithControl g15b(P0Pu = P0Pu_g15b, Q0Pu = Q0Pu_g15b, U0Pu = 1.0455, UPhase0 = SIunits.Conversions.from_deg(-52.19), gen = GeneratorWithControl.GeneratorParameters.genFramePreset.g15) annotation(
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Grid/FullDynamicModelB.mo:44]: "gen =  " to be checked

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Grid/FullDynamicModelB.mo [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  // P0Pu, Q0Pu in SnRef, receptor convention
+  final parameter Types.ActivePowerPu P0Pu_g18a = -530.0 / Electrical.SystemBase.SnRef;
+  final parameter Types.ReactivePowerPu Q0Pu_g18a = -110.5 / Electrical.SystemBase.SnRef;
+  // Generator g18b init values:
+  // P0Pu, Q0Pu in SnRef, receptor convention
+  final parameter Types.ActivePowerPu P0Pu_g18b = -530.0 / Electrical.SystemBase.SnRef;
+  final parameter Types.ReactivePowerPu Q0Pu_g18b = -110.5 / Electrical.SystemBase.SnRef;
+  // Generator g20 init values:
+  // P0Pu, Q0Pu in SnRef, receptor convention
+  final parameter Types.ActivePowerPu P0Pu_g20b = -1537.4 / Electrical.SystemBase.SnRef;
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Grid/FullDynamicModelB.mo:78]: We should check that this is 600MW less than case A

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Grid/FullDynamicModelB.mo [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+model FullDynamicModelB "Nordic test grid with buses, lines, shunts, voltage-dependent loads, transformers and generators (operating point B)"
+  import Modelica.SIunits;
+  import Dynawo;
+  import Dynawo.Examples.Nordic.Components.GeneratorWithControl;
+  import Dynawo.Types;
+
+  extends Dynawo.Examples.Nordic.Grid.FullDynamicModelA(
+    g15(P0Pu = P0Pu_g15a, Q0Pu = Q0Pu_g15a),
+    g16(P0Pu = P0Pu_g16a, Q0Pu = Q0Pu_g16a),
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Grid/FullDynamicModelB.mo:24]: Modifs on g15 and g18 not necessary

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Components/GeneratorWithControl/GeneratorParameters.mo [hunk 5]
```diff
@@ -15,7 +15,7 @@ within Dynawo.Examples.Nordic.Components.GeneratorWithControl;
 
 record GeneratorParameters "Parameter sets for the generators of the Nordic 32 test system"
 
-  type genFramePreset = enumeration(g01, g02, g03, g04, g05, g06, g07, g08, g09, g10, g11, g12, g13, g14, g15, g16, g17, g18, g19, g20) "Generator names";
+  type genFramePreset = enumeration(g01, g02, g03, g04, g05, g06, g07, g08, g09, g10, g11, g12, g13, g14, g15, g15a, g15b, g16, g16b, g17, g18, g18a, g18b, g19, g20) "Generator names";
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Components/GeneratorWithControl/GeneratorParameters.mo:18]: Modifs on g15 and g18 not necessary

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Components/GeneratorWithControl/GeneratorParameters.mo [hunk 6]
```diff
@@ -15,7 +15,7 @@ within Dynawo.Examples.Nordic.Components.GeneratorWithControl;
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Components/GeneratorWithControl/GeneratorParameters.mo:15]: This file should in fact remain the same

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Grid/FullDynamicModelB.mo [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    g18(P0Pu = P0Pu_g18a, Q0Pu = Q0Pu_g18a),
+    g20(P0Pu = P0Pu_g20b),
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Grid/FullDynamicModelB.mo:26]: This is the only modif that should remain

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Grid/FullDynamicModelB.mo [hunk 8]
```diff
@@ -0,0 +1,112 @@
+within Dynawo.Examples.Nordic.Grid;
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Grid/FullDynamicModelB.mo:1]: Please check init

**Review by cossartque — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Components/GeneratorWithControl/GeneratorParameters.mo (hunk 5 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Components/GeneratorWithControl/GeneratorParameters.mo:18]: Modifications canceled

File: dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Grid/FullDynamicModelB.mo (hunk 4 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Examples/Nordic/Grid/FullDynamicModelB.mo:24]: Modifications reversed

**Pull request closed without merging**